homeassistant/components/bosch_shc/config_flow.py

```diff
@@ -187,16 +187,23 @@ async def async_step_zeroconf(self, discovery_info):
             return self.async_abort(reason="not_bosch_shc")
 
         try:
-            self.info = info = await self._get_info(discovery_info["host"])
+            for host in (
+                discovery_info["host"]
+                if isinstance(discovery_info["host"], list)
+                else [discovery_info["host"]]
+            ):
+                if not host.startswith("169."):
+                    self.info = info = await self._get_info(host)
+                    self.host = host
+                    break
         except SHCConnectionError:
             return self.async_abort(reason="cannot_connect")
 
         local_name = discovery_info["hostname"][:-1]
         node_name = local_name[: -len(".local")]
 
         await self.async_set_unique_id(info["unique_id"])
-        self._abort_if_unique_id_configured({CONF_HOST: discovery_info["host"]})
-        self.host = discovery_info["host"]
```

Comment:
Why remove this line?

Reply:
`self.host` is already assigned above after testing if the address is a link local/dynamically assigned one.
`self._abort_if_unique_id_configured` checks the host, not the return of the discovery info value.